tests/Tests.AzureAppConfiguration/Integration/IntegrationTests.cs

```diff
@@ -0,0 +1,544 @@
+        {
+            string keyPrefix = GetUniqueKeyPrefix(testName);
+            string sentinelKey = $"{keyPrefix}:Sentinel";
+            string featureFlagKey = $".appconfig.featureflag/{keyPrefix}Feature";
+
+            // Create test-specific settings
+            var testSettings = new List<ConfigurationSetting>
+            {
+                new ConfigurationSetting($"{keyPrefix}:Setting1", "InitialValue1"),
+                new ConfigurationSetting($"{keyPrefix}:Setting2", "InitialValue2"),
+                new ConfigurationSetting(sentinelKey, "Initial"),
+                ConfigurationModelFactory.ConfigurationSetting(
+                    featureFlagKey,
+                    @"{""id"":""" + keyPrefix + @"Feature"",""description"":""Test feature"",""enabled"":false}",
+                    contentType: FeatureManagementConstants.ContentType)
+            };
+
+            // Add test-specific settings to the store
+            foreach (var setting in testSettings)
+            {
+                await _configClient.SetConfigurationSettingAsync(setting);
+            }
+
+            return (keyPrefix, sentinelKey, featureFlagKey);
+        }
+
+        [Fact]
+        public void LoadConfiguration_RetrievesValuesFromAppConfiguration()
+        {
+            Skip.If(_skipTests, _skipReason);
+
+            // Arrange & Act
+            var config = new ConfigurationBuilder()
+                .AddAzureAppConfiguration(options =>
+                {
+                    options.Connect(GetConnectionString());
+                    options.Select($"{TestKeyPrefix}:*");
+                })
+                .Build();
+
+            // Assert
+            Assert.Equal("InitialValue1", config[$"{TestKeyPrefix}:Setting1"]);
+            Assert.Equal("InitialValue2", config[$"{TestKeyPrefix}:Setting2"]);
+        }
+
+        [Fact]
+        public async Task TryRefreshAsync_UpdatesConfiguration_WhenSentinelKeyChanged()
+        {
+            Skip.If(_skipTests, _skipReason);
+
+            // Arrange - Setup test-specific keys
+            var (keyPrefix, sentinelKey, _) = await SetupTestKeys("UpdatesConfig");
+            IConfigurationRefresher refresher = null;
+
+            var config = new ConfigurationBuilder()
+                .AddAzureAppConfiguration(options =>
+                {
+                    options.Connect(GetConnectionString());
+                    options.Select($"{keyPrefix}:*");
+                    options.ConfigureRefresh(refresh =>
+                    {
+                        refresh.Register(sentinelKey, refreshAll: true)
+                              .SetRefreshInterval(TimeSpan.FromSeconds(1));
+                    });
+
+                    refresher = options.GetRefresher();
+                })
+                .Build();
+
+            // Verify initial values
+            Assert.Equal("InitialValue1", config[$"{keyPrefix}:Setting1"]);
```

Comment:
I think you should have one or multiple pre-cache expiration calls to ensure that the keys aren't refreshed until the cache expiration is up. That way cache expiration both ways can be tested here.